purchases/src/main/kotlin/com/revenuecat/purchases/PurchaseParams.kt

```diff
@@ -136,6 +136,10 @@ data class PurchaseParams(val builder: Builder) {
         }
 
         open fun build(): PurchaseParams {
+            product?.let {
+                ensureNoTestProduct(it)
```

Comment:
I moved this to `build` so that it doesn't have to run during the new tests. Instead, we mock the call to `purchase` with the builder. That way we can still test the rest without it failing to build purchase params with a test product.